src/pages/ServiceDetails/ServiceInfo/types/DestinationRule.ts

```diff
@@ -0,0 +1,70 @@
+import { DestinationRuleSpec, K8sMetadata } from '../../../../types/IstioObjects';
+import Subset from './Subset';
+
+export default class DestinationRule {
```

Comment:
We are re-reclaring the same DestinationRule in a different place.
So we will have one in IstioObjects.ts and another here to be used for same purpose, I think.

Reply:
I guess that is not the same purpose. The purpose in the IstioObjects is just to have an interface with no logic attached to it. The reason I added a class here is to distribute some logic that are just unique for this components. The validations of format are built depending on which information is shown in the list of this component. It wouldn't be the same in other components (e.g. IstioConfigList/Details).
One more reason I add this class here is not to perform all the logic in the same component. To distribute it between different classes (and try to achieve a one single responsibility class).
Perhaps, instead of having a DestinationRule class, it would be better to have a DestinationRuleValidator.ts where it takes a DR and computes the whole validation.